### PR TITLE
Administrationhome

### PR DESCRIPTION
**Pull request by ThomasDCW**
On product backlog branch 021-administration-home

Install React admin, for now there is a page where we get list for users, topics, lessons

NEXT step auth in react admin 
cf Backlog -> 022-administration-authentification
I guess we need a user who get the right in db or somewhere...

I had an error in console when i use import "@servicec/api" so i was obliged to use "../services/api"

### CODE REVIEW

**Review by jeremyjorand — APPROVED** [review]
ok for me

File: admin/src/dataProvider.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { fetchUtils } from "react-admin";
+import api from "../services/api";
```
Comment (LuigiMenez) [inline admin/src/dataProvider.js:2]: Use import absolute

File: admin/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { defineConfig } from 'vite';
+const path = require('path');
```
Comment (LuigiMenez) [inline admin/vite.config.js:2]: You can't use the imports instead of the require?

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
Use import absolute

**Review by jeremyjorand — APPROVED** [review]
ok for me

**Review by LuigiMenez — APPROVED** [review]
Ok for me

**Review by Dash293 — APPROVED** [review]

File: admin/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import dataProvider from "./dataProvider";
+
+function App() {
+  return (
+    <div className="App">
+      <Admin dataProvider={dataProvider}>
+        <Resource name="users" list={ListGuesser} />
+        <Resource name="lessons" list={ListGuesser} />
+        <Resource name="topics" list={ListGuesser} />
```
Comment (LoicBrassart) [inline admin/src/App.jsx:11]: You'll have to rewrite these *Guessers later

File: admin/src/authProvider.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+export default {
+  login: async (params) => {
+    try {
+      const formData = {
+        email: params.username,
+        password: params.password,
+      };
+      const { data } = await api.post(`/auth/login`, formData);
+      localStorage.setItem("token", data.token);
```
Comment (LoicBrassart) [inline admin/src/authProvider.js:11]: NB: You know how to register a token in a LocalStorage (as an answer to this morning's stand-up)

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**